src/runtime/threads/detail/thread_num_tss.cpp

```diff
@@ -7,17 +7,24 @@
 #include <hpx/runtime/threads/detail/thread_num_tss.hpp>
 
 #include <cstddef>
-#include <utility>
+#include <cstdint>
+#include <tuple>
 
-namespace hpx { namespace threads { namespace detail
-{
+namespace hpx { namespace threads { namespace detail {
     namespace {
         std::size_t& thread_num_tss()
         {
             HPX_NATIVE_TLS std::size_t thread_num_tss_ = std::size_t(-1);
             return thread_num_tss_;
         }
-    }
+
+        thread_tuple& thread_numbers_tss()
+        {
+            HPX_NATIVE_TLS thread_tuple thread_numbers_tss_ = {
+                std::uint16_t(-1), std::uint16_t(-1)};
+            return thread_numbers_tss_;
+        }
+    }    // namespace
```

Comment:
actually, the thread number and the pool id. it turns out the thread indexing was changed/broken at some point during recent changes and so I need the pool index to correctly get the local/global thread numbering right. 
I thought about removing the old thread_num_tss but decided that it'd be better to leave the existing stuff as-is and after this has been used for a while and is considered safe/stable to thn do a cleanup in the next version

Reply:
Then we can maybe call them that? And I guess there's no real need to pack them into a tuple instead of having them separate like the global thread index. The tuple has no names for the elements so it's hard to know what's what.

Reply:
I didn't want to use extra bytes so I packed 2x16bit nums into a tuple. The existing use of the thread_num stuff was totally undocumented, so I thought I could sneak the pool Id in there too without any objection. I'd like to leave the tuple as-is,  but I'll add some comments in the code to explain what's going on

Reply:
I added some comments. it's local thread id and pool id, the other one is global thread id, so they actually complement each other.

Reply:
The comments in the `.cpp` file say global thread num and pool id, while the `.hpp` file says local thread num and pool id. I guess it's actually the latter? (Note: please don't push any changes now so that we see what the CI says about the latest changes.)

Reply:
It should be local thread num.